extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/HibernateMetadata.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.hibernate.orm;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * CDI bean that contains useful Hibernate metadata
+ *
+ * The bean will only be present if the following conditions apply
+ *
+ * <p>
+ * <ul>
+ * <li>There is at least one entity
+ * <li>There exists at least one CDI injection point for the class
+ * </ul>
+ * <p>
+ *
+ * Warning: this API is experimental and subject to change as the Quarkus / Hibernate integration evolves
+ */
+public interface HibernateMetadata {
+
+    String DEFAULT_PERSISTENCE_UNIT_NAME = "default";
+
+    /**
+     * Returns a map where the persistence unit name is the key and the class names of the associated entities are the value
+     */
+    Map<String, Set<String>> entitiesByPersistenceUnit();
+
+    /**
+     * Returns a set of class names of the entities that are associated with the default persistence unit
+     */
+    default Set<String> entitiesOfDefaultPersistenceUnit() {
+        return entitiesByPersistenceUnit().getOrDefault(DEFAULT_PERSISTENCE_UNIT_NAME, Collections.emptySet());
```

Comment:
I was wondering if a `PersistenceUnit` container would be useful instead?

You would have `getPersistenceUnit(String)` and `getDefaultPersistenceUnit()` which would return a `PersistenceUnit` and you would have a `getEntities()` in this container.

Reply:
Interesting, I like it.

I assume you that the new class would be called `PersistenceUnitMetadata` or something?

Reply:
PR updated to reflect this design